packages/actor-query-result-serialize-stats/lib/ActorQueryResultSerializeStats.ts

```diff
@@ -68,9 +68,13 @@ export class ActorQueryResultSerializeStats extends ActorQueryResultSerializeFix
 
     const startTime = this.now();
     let result = 1;
+
+    function* end(cb: () => string): Generator<string> {
+      yield cb();
+    }
     const stream = wrap(resultStream)
       .map(() => this.createStat(startTime, result++))
-      .append([ this.createFooter(startTime) ]);
+      .append(wrap(end(() => this.createFooter(startTime))));
```

Comment:
Have you manually confirmed that this produces the expected output now?

Because I would expect that `wrap` creates an asynciterator with `autoStart: true`, and therefore eagerly creates the footer before it's actually needed.
But I could be incorrect of course.

Reply:
It should use the [`WrappingIterator`](https://github.com/RubenVerborgh/AsyncIterator/blob/266e5deb20df6d9033f01c137bcafee6721b1215/asynciterator.ts#L2223) which has no buffering and thus no autostart, instead pulling elements on-demand.

Will double check with a `stats` integration test.

Reply:
Aha, then I will probably be wrong indeed.